### PR TITLE
Get Directory Blocks From Persistence

### PR DESCRIPTION
**Pull request by KaustubhShamshery**
## Description
Instead of retrieving just DS Blocks all VC/FB/DS should be retrieved

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [x] small-scale cloud test

### CODE REVIEW

File: src/libPersistence/Retriever.cpp [hunk 1]
```diff
@@ -167,6 +167,171 @@ void Retriever::RetrieveTxBlocks(bool& result, const bool& wakeupForUpgrade) {
...
+        LOG_GENERAL(WARNING, "Could not delete VC block");
+      }
+    } else if (std::get<BlockLinkIndex::BLOCKTYPE>(blocklink) ==
+               BlockType::FB) {
+      if (!BlockStorage::GetBlockStorage().DeleteFallbackBlock(
+              std::get<BlockLinkIndex::BLOCKHASH>(blocklink))) {
+        LOG_GENERAL(WARNING, "Could not deleteLoop  FB block");
+      }
+    }
+  }
```
Comment (ckyang) [inline src/libPersistence/Retriever.cpp]: When toDelete is true, it means not only blockstorage should be removed, but also new block should not be added to m_blocklinkchain. We should consider toDelete before calling m_blocklinkchain.AddBlockLink().

File: src/libNode/Node.cpp [hunk 2]
```diff
@@ -233,6 +233,20 @@ bool Node::StartRetrieveHistory(bool& wakeupForUpgrade) {
...
+  {
+    lock_guard<mutex> lock(m_mediator.m_mutexInitialDSCommittee);
+    if (m_mediator.m_initialDSCommittee->size() != 0) {
+      for (const auto& initDSCommKey : *m_mediator.m_initialDSCommittee) {
+        m_mediator.m_blocklinkchain.GetBuiltDSComm().emplace_back(initDSCommKey,
+                                                                  Peer());
+        // Set initial ds committee with null peer
+      }
+    } else {
+      LOG_GENERAL(WARNING, "Initial DS comm size 0 ");
```
Comment (bb111189) [inline src/libNode/Node.cpp]: I think this should be upgraded to FATAL as initial DS comm cannot be 0

File: src/libPersistence/BlockStorage.cpp [hunk 3]
```diff
@@ -408,6 +418,36 @@ bool BlockStorage::GetAllTxBodiesTmp(std::list<TxnHash>& txnHashes) {
...
+      LOG_GENERAL(WARNING, "Lost one blocklink in the chain");
+      delete it;
+      return false;
+    }
+    BlockLink blcklink;
+    if (!Messenger::GetBlockLink(
+            vector<unsigned char>(blockString.begin(), blockString.end()), 0,
+            blcklink)) {
+      LOG_GENERAL(WARNING, "Deserialization of blockLink failed " << bns);
+      return false;
```
Comment (bb111189) [inline src/libPersistence/BlockStorage.cpp]: need to delete it here?

File: src/libPersistence/Retriever.cpp [hunk 4]
```diff
@@ -167,6 +167,172 @@ void Retriever::RetrieveTxBlocks(bool& result, const bool& wakeupForUpgrade) {
...
+      LOG_GENERAL(INFO, "Has incompleted DS Block, remove it");
+      toDelete = true;
+    }
+  }
+
+  uint64_t lastDsIndex = std::get<BlockLinkIndex::DSINDEX>(blocklinks.back());
+  const BlockType lastType =
+      std::get<BlockLinkIndex::BLOCKTYPE>(blocklinks.back());
+  if (lastType != BlockType::DS) {
+    lastDsIndex--;
```
Comment (bb111189) [inline src/libPersistence/Retriever.cpp]: Should check for underflow

**Pull request merged**